### PR TITLE
Modernize template

### LINKED ISSUE #29: More flake8 plugins
**Expected behaviour**
The development of the new plugin should be as easy as possible. Some of this can be achieved with a good set of Flake8 plugins. 

**Current behaviour**
There are few plugins, but there could always be more...

**Additional context**
Here is a list of some of the useful plugins:

* flake8-print
* flake8-tidy-imports
* flake8-comprehensions
* flake8-simplify
* flake8-pytest-style
* flake8-pie

### PR DESCRIPTION
**Pull request by LKajan**
Main changes:

- Change to use Ruff as a linter and formatter
- Fixes deprecated VS Code settings

Resolves #29

### CODE REVIEW

**Comment by JuhoErvasti** [comment]
Not sure if this is caused by anything in the implementation itself, but on Ubuntu in VS Code mypy is reporting missing library stubs in the plugin.py file for all imports from the qgis package.

```
Skipping analyzing "qgis.core": module is installed, but missing library stubs or py.typed markerMypy[import-untyped]
```

Similar error in build.py:

```
Cannot find implementation or library stub for module named "qgis_plugin_tools.infrastructure.plugin_maker"
```

**Review by nmaarnio — APPROVED** [review]
Looks good to me, although I might have missed something in my testing since I did not get the same error as @JuhoErvasti

File: {{cookiecutter.project_directory}}/{{cookiecutter.plugin_package}}/{{cookiecutter.plugin_package}}_processing/provider.py.j2 [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from qgis.core import QgsProcessingProvider
 
-from .processing_algorithm import ProcessingAlgorithm
+from {{cookiecutter.plugin_package}}.processing.processing_algorithm import ProcessingAlgorithm
```
Comment (JuhoErvasti) [inline {{cookiecutter.project_directory}}/{{cookiecutter.plugin_package}}/{{cookiecutter.plugin_package}}_processing/provider.py.j2:3]: Adress https://github.com/GispoCoding/cookiecutter-qgis-plugin/pull/46#issuecomment-1938250661

```suggestion
from {{cookiecutter.plugin_package}}.{{cookiecutter.plugin_package}}_processing.processing_algorithm import ProcessingAlgorithm
```

Comment (LKajan) [reply {{cookiecutter.project_directory}}/{{cookiecutter.plugin_package}}/{{cookiecutter.plugin_package}}_processing/provider.py.j2]: This should now be addressed

**Comment by LKajan** [comment]
> Not sure if this is caused by anything in the implementation itself, but on Ubuntu in VS Code mypy is reporting missing library stubs in the plugin.py file for all imports from the qgis package.
> 
> ```
> Skipping analyzing "qgis.core": module is installed, but missing library stubs or py.typed markerMypy[import-untyped]
> ```
> 
> Similar error in build.py:
> 
> ```
> Cannot find implementation or library stub for module named "qgis_plugin_tools.infrastructure.plugin_maker"
> ```

I ignored those missing type errors on qgis and osgeo package. And for qgis_plugin_tools all typing errors completly.

**Pull request merged**